### PR TITLE
Dev

### PR DESCRIPTION
**Pull request by Asafv**

### CODE REVIEW

**Comment by MaTriXy** [comment]
Hi,
I would like you to change the volley usage to be with Retrofit2 and okhttp3.

File: gradle/wrapper/gradle-wrapper.properties [hunk 1]
```diff
@@ -0,0 +1,6 @@
+#Mon Dec 28 10:00:20 PST 2015
+distributionBase=GRADLE_USER_HOME
+distributionPath=wrapper/dists
+zipStoreBase=GRADLE_USER_HOME
+zipStorePath=wrapper/dists
+distributionUrl=https\://services.gradle.org/distributions/gradle-2.14.1-all.zip
```
Comment (MaTriXy) [inline gradle/wrapper/gradle-wrapper.properties]: Gradle should be 3.3
http://services.gradle.org/distributions/gradle-3.3-all.zip

File: app/src/main/res/layout/movie_grid_item.xml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:layout_width="wrap_content"
+    android:layout_height="wrap_content"
+    android:gravity="center"
+    android:background="@color/light_gray"
+    android:orientation="vertical"
+    android:padding="5dp">
+
+    <TextView
+        android:id="@+id/tv_movie_title"
```
Comment (MaTriXy) [inline app/src/main/res/layout/movie_grid_item.xml]: move sizes and properties to style.

File: app/src/main/java/com/asafvaron/themoviedbtest/objects/MyMovie.java [hunk 3]
```diff
@@ -0,0 +1,96 @@
+package com.asafvaron.themoviedbtest.objects;
+
+/**
+ * Created by asafvaron on 19/02/2017.
+ */
+public class MyMovie {
```
Comment (MaTriXy) [inline app/src/main/java/com/asafvaron/themoviedbtest/objects/MyMovie.java]: I would like you to use Gson or Jackson as well. 
Get used to Using annotations.

File: app/src/main/java/com/asafvaron/themoviedbtest/adapters/MoviesGridAdapter.java [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    private void init() {
+        mRvGridList.setHasFixedSize(true);
+        mRvGridList.setLayoutManager(new GridLayoutManager(mContext, 2));
+        mRvGridList.setItemAnimator(new DefaultItemAnimator());
+        // set itemDecoration for grid spacing
+        GridSpacingItemDecoration decoration = new GridSpacingItemDecoration(mContext, R.dimen.grid_item_offset);
+        mRvGridList.addItemDecoration(decoration);
+        mRvGridList.setAdapter(this);
+        this.notifyDataSetChanged();
```
Comment (MaTriXy) [inline app/src/main/java/com/asafvaron/themoviedbtest/adapters/MoviesGridAdapter.java]: bad practice... 
When using recyclerview you should almost never use notifyDataSetChanged. 
Also, Why hold all the references inside the adapter?

File: app/src/main/java/com/asafvaron/themoviedbtest/MyApp.java [hunk 5]
```diff
@@ -1,22 +1,28 @@
...
+    private static Context mContext;
+
     @Override
     public void onCreate() {
         super.onCreate();
 
-        // testing MoviesApiEnum
-
-//        MoviesApi.getInstance().init(getApplicationContext());
+        mContext = getApplicationContext();
```
Comment (MaTriXy) [inline app/src/main/java/com/asafvaron/themoviedbtest/MyApp.java:21]: Why is this needed?

**Comment by Asafv** [comment]
Please explain regarding the references inside the adapter, does it impact Performance/Memory or just bad coding practice?

Thanks

**Comment by Asafv** [comment]
Do you want me to use both in the project?

Thanks

**Comment by Asafv** [comment]
I removed it from code, Me and Shahar had a debate regarding the usage of enum as a singleton.

File: app/src/main/res/values/styles.xml [hunk 6]
```diff
@@ -8,4 +8,54 @@
...
+    <style name="MovieInfoImageStyle">
+        <item name="android:layout_width">240dp</item>
+        <item name="android:cropToPadding">true</item>
+        <item name="android:layout_alignParentStart">true</item>
+        <item name="android:layout_alignParentLeft">true</item>
+        <item name="android:layout_height">300dp</item>
+    </style>
+
+    <style name="InfoTextViewStyle">
+        <item name="android:layout_width">wrap_content</item>
```
Comment (MaTriXy) [inline app/src/main/res/values/styles.xml]: Width and height are not needed within style if they are wrap_content or match_parent.
Style for width and height is used for special sizes...

File: app/src/main/res/layout/movie_info_layout.xml [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    <TextView
+        android:id="@+id/tv_runtime"
+        style="@style/InfoTextViewStyle"
+        android:layout_below="@+id/tv_vote_average"
+         />
+
+    <TextView
+        android:id="@+id/tv_overview"
+        android:layout_width="match_parent"
```
Comment (MaTriXy) [inline app/src/main/res/layout/movie_info_layout.xml]: Fix by using dimens

File: app/src/main/java/com/asafvaron/themoviedbtest/rest/ApiInterface.java [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+import retrofit2.http.GET;
+import retrofit2.http.Path;
+import retrofit2.http.Query;
+
+/**
+ * Created by asafvaron on 20/02/2017.
+ */
+public interface ApiInterface {
+
+    @GET("movie/top_rated?api_key=a2ddc78251984ac1b6853aab3885b44c")
```
Comment (MaTriXy) [inline app/src/main/java/com/asafvaron/themoviedbtest/rest/ApiInterface.java]: Why is api_key hardcoded?

File: app/src/main/java/com/asafvaron/themoviedbtest/fragments/MovieInfoFragment.java [hunk 9]
```diff
@@ -0,0 +1,54 @@
...
+        if (getArguments()!=null)
+            mMovie = (Movie) getArguments().getSerializable("movie");
+    }
+
+    @Nullable
+    @Override
+    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
+        View root = inflater.inflate(R.layout.movie_info_layout, container, false);
+
+        Glide.with(getActivity())
```
Comment (MaTriXy) [inline app/src/main/java/com/asafvaron/themoviedbtest/fragments/MovieInfoFragment.java]: Check for reusing the url from the movie grid view so it won't need to reload from network and you might be able to have a nice transition later on...
